### ISSUE TITLE
cvat_server cannot connect cvat_db

### ISSUE THREAD

**Issue by HanSHANGOM** [cvat-ai/cvat#6880]
![image](https://github.com/opencv/cvat/assets/34465203/8ea2e6ef-1c0e-45c4-b847-15b637d7bcfc)
![image](https://github.com/opencv/cvat/assets/34465203/9bb7ebad-1600-4bba-b8c9-2df1aed3c0e7)
![image](https://github.com/opencv/cvat/assets/34465203/10bd4261-df83-4449-8ac1-0f6500a6942a)
```
Traceback (most recent call last):
  File "/home/django/manage.py", line 20, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/opt/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/venv/lib/python3.10/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/venv/lib/python3.10/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
  File "/opt/venv/lib/python3.10/site-packages/django/core/management/base.py", line 106, in wrapper
    res = handle_func(*args, **kwargs)
  File "/opt/venv/lib/python3.10/site-packages/django/core/management/commands/migrate.py", line 117, in handle
    executor = MigrationExecutor(connection, self.migration_progress_callback)
  File "/opt/venv/lib/python3.10/site-packages/django/db/migrations/executor.py", line 18, in __init__
    self.loader = MigrationLoader(self.connection)
  File "/opt/venv/lib/python3.10/site-packages/django/db/migrations/loader.py", line 58, in __init__
    self.build_graph()
  File "/opt/venv/lib/python3.10/site-packages/django/db/migrations/loader.py", line 235, in build_graph
    self.applied_migrations = recorder.applied_migrations()
  File "/opt/venv/lib/python3.10/site-packages/django/db/migrations/recorder.py", line 81, in applied_migrations
    if self.has_table():
  File "/opt/venv/lib/python3.10/site-packages/django/db/migrations/recorder.py", line 57, in has_table
    with self.connection.cursor() as cursor:
  File "/opt/venv/lib/python3.10/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/base/base.py", line 330, in cursor
    return self._cursor()
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/base/base.py", line 306, in _cursor
    self.ensure_connection()
  File "/opt/venv/lib/python3.10/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/base/base.py", line 288, in ensure_connection
    with self.wrap_database_errors:
  File "/opt/venv/lib/python3.10/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/base/base.py", line 289, in ensure_connection
    self.connect()
  File "/opt/venv/lib/python3.10/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/base/base.py", line 270, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/opt/venv/lib/python3.10/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/postgresql/base.py", line 275, in get_new_connection
    connection = self.Database.connect(**conn_params)
  File "/opt/venv/lib/python3.10/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
django.db.utils.OperationalError: connection to server at "cvat_db" (192.168.224.3), port 5432 failed: fe_sendauth: no password supplied

wait-for-it.sh: waiting for cvat_db:5432 without a timeout
wait-for-it.sh: cvat_db:5432 is available after 0 seconds
Traceback (most recent call last):
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/base/base.py", line 289, in ensure_connection
    self.connect()
  File "/opt/venv/lib/python3.10/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/base/base.py", line 270, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/opt/venv/lib/python3.10/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.10/site-packages/django/db/backends/postgresql/base.py", line 275, in get_new_connection
    connection = self.Database.connect(**conn_params)
  File "/opt/venv/lib/python3.10/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: connection to server at "cvat_db" (192.168.224.3), port 5432 failed: fe_sendauth: no password supplied
```

these are All I can show you. I tried to find what is wrong, but i couldn't. i suffered this on v2.7.1 and also v2.4.9 ..

so I am using V2.2.0.. could you please help me to resolve this problem?

**Comment by azhavoro** [comment]
How we can reproduce the issue?

**Comment by sanghun-scalawox** [comment]
> How we can reproduce the issue?

I don't know. someday suddenly it happened..

**Comment by jelly-Ding** [comment]
@sanghun-scalawox Hello. Do you have any idea how to fix this?  I'm getting this same issue. All docker services are working well

**Comment by HanSHANGOM** [comment]
@jelly-Ding Hello, I don't know how. I had backup for db, pruned and reinstalled all docker images and volumes, restored db. but it was happend again. so I concluded that cvat_db image has the problem.. 
and now I do not docker compose up with docker-compose.dev.yml.. when I do command without this yml file, it seems not reproduced..(I guess)
but I think it can be happend anytime. because I don't know that(excluding docker-compose.dev.yml) is actually solution about cvat_db problem.

**Comment by Crotonix** [comment]
Not sure if this issue has been resolved but this is a persistent issue. I recently upgraded cvat from 2.7.4 -> 2.16.3. I never faced this issue on 2.7.4(using for about 6 months on remote). But after migration I faced this issue within 4 days even when no annotation work is being done and server is idle after upgrade. Since I had backup and no new data was added I upgraded it again to 2.17.0. It worked fine till yesterday(about 30 days) where the db image is corrupted and will not spin up the server. Only difference between 2.7.4 to 2.17.0 was that the later was composed up with server less function to support automatic annotation. 
```sh
docker exec -t cvat_server python manage.py health_check
```
returns 
'''sh
Error response from daemon: Container d13d60a8bbed447eb1de5fe31ffa61a6469fa01724ed13f2f840e7e3ab34eca9 is restarting, wait until the container is running
```
I will help to reproduce it but it has something to do with postgres. What it does is that when you access the remote server ui it displays two errors : 
![image](https://github.com/user-attachments/assets/23384765-5950-4bb5-a21a-33603770447a)
Attaching logs:
[cvat_server.log](https://github.com/user-attachments/files/17083199/cvat_server.log)
[cvat_db.log](https://github.com/user-attachments/files/17083200/cvat_db.log)
